src/rgw/rgw_rest_s3.cc

```diff
@@ -5870,3 +5875,322 @@ bool rgw::auth::s3::S3AnonymousEngine::is_applicable(
 
   return route == AwsRoute::QUERY_STRING && version == AwsVersion::UNKNOWN;
 }
+
+
+using namespace s3selectEngine;
+const char* RGWSelectObj_ObjStore_S3::header_name_str[3] = {":event-type", ":content-type", ":message-type"};
+const char* RGWSelectObj_ObjStore_S3::header_value_str[3] = {"Records", "application/octet-stream", "event"};
+
+RGWSelectObj_ObjStore_S3::RGWSelectObj_ObjStore_S3()
+{
+  s3select_syntax = std::make_unique<s3selectEngine::s3select>(),
+  m_s3_csv_object = std::unique_ptr<s3selectEngine::csv_object>(),
+  m_buff_header = std::make_unique<char[]>(1000),
+  chunk_number = 0,
+  crc32 = std::unique_ptr<boost::crc_32_type>(),
+
+  set_get_data(true);
+}
+
+RGWSelectObj_ObjStore_S3::~RGWSelectObj_ObjStore_S3()
+{
+}
+
+int RGWSelectObj_ObjStore_S3::get_params()
+{
+
+  //retrieve s3-select query from payload
+  bufferlist data;
+  int ret;
+  int max_size = 4096;
+  std::tie(ret, data) = rgw_rest_read_all_input(s, max_size, false);
+  if (ret != 0)
+  {
+    ldout(s->cct, 10) << "s3-select query: failed to retrieve query; ret = " << ret << dendl;
+    return ret;
+  }
+
+  m_s3select_query = data.to_str();
+  if (m_s3select_query.length() > 0)
+  {
+    ldout(s->cct, 10) << "s3-select query: " << m_s3select_query << dendl;
+  }
+  else
+  {
+    ldout(s->cct, 10) << "s3-select query: failed to retrieve query;" << dendl;
+    return -1;
+  }
```

Comment:
please be consistent with the project's [CodingStyle](https://github.com/ceph/ceph/blob/master/CodingStyle) throughout:
```suggestion
  if (m_s3select_query.length() > 0) {
    ldout(s->cct, 10) << "s3-select query: " << m_s3select_query << dendl;
  } else {
    ldout(s->cct, 10) << "s3-select query: failed to retrieve query;" << dendl;
    return -1;
  }
```

Reply:
hm--why did the gnu indent pass miss that?

Reply:
when it comes to brackets, I think that consistency of style should be the guideline.
so, in the rgw_... files we should follow the K&r style that we have there already.
but in the s3select files we can keep the BSD style indentation.

Reply:
I do not see a reason for any inconsistency

Reply:
`rgw_rest_s3.cc ` is using k&r indentation, so we should probably keep it with any new code
the code from the separate repo can use any other indentation style

Reply:
I agree about not gratuitously respacing code in a whole file, but if a file is out of step with other code, I do expect it it to gradually modernize, or would be happy if that happens;  I requested gal change some style things in both repos, though, and gal's suggestion to use gnu indent with a consistent set of options seemed like a good idea.  I hope we can stick with that plan.